tests/src/kakarot/accounts/test_account_contract.py

```diff
@@ -351,176 +348,508 @@ def test_should_execute_starknet_call(self, cairo_run):
-                r=int_to_uint256(signed.r),
-                s=int_to_uint256(signed.s),
-                v=signed["v"],
-                tx_data=tx_data,
-            )
+            with cairo_error(message="Invalid signature."), SyscallHandler.patch(
+                "Account_evm_address", address
+            ):
+                cairo_run(
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=CHAIN_ID + 1,
+                )
 
-        @pytest.mark.parametrize("transaction", TRANSACTIONS)
-        async def test_should_raise_with_wrong_signed_chain_id(
-            self, cairo_run, transaction
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        def test_should_raise_invalid_chain_id_tx_type_different_from_0(
+            self, cairo_run
         ):
+            transaction = {
+                "type": 2,
+                "gas": 100_000,
+                "maxFeePerGas": 2_000_000_000,
+                "maxPriorityFeePerGas": 2_000_000_000,
+                "data": "0x616263646566",
+                "nonce": 34,
+                "to": "",
+                "value": 0x00,
+                "accessList": (
+                    {
+                        "address": "0x0000000000000000000000000000000000000001",
+                        "storageKeys": (
+                            "0x0100000000000000000000000000000000000000000000000000000000000000",
+                        ),
+                    },
+                ),
+                "chainId": CHAIN_ID,
+            }
+            encoded_unsigned_tx = rlp_encode_signed_data(transaction)
+            tx_data = list(encoded_unsigned_tx)
             private_key = generate_random_private_key()
             address = int(private_key.public_key.to_checksum_address(), 16)
             signed = Account.sign_transaction(transaction, private_key)
+            signature = [
+                *int_to_uint256(signed.r),
+                *int_to_uint256(signed.s),
+                signed["v"],
+            ]
+
+            with cairo_error(message="Invalid chain id"), SyscallHandler.patch(
+                "Account_evm_address", address
+            ):
+                cairo_run(
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=CHAIN_ID + 1,
+                )
 
+        @SyscallHandler.patch(
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
+        )
+        @SyscallHandler.patch(
+            "Account_nonce",
+            100,
+        )
+        @pytest.mark.parametrize("transaction", TRANSACTIONS)
+        def test_should_raise_invalid_nonce(self, cairo_run, transaction):
```

Comment:
I think that this test passes because default nonce is 0 (not patched) while the TRANSACTIONS have nonce > 0. But it's better to actually patch the `Account_nonce` and explicity put the nonce to something else in the `transaction`